### ISSUE TITLE
As SCT team, we would like dapp-manager to regain some time-performance budget (be faster)

### ISSUE THREAD

**Issue by grisha87** [golemfactory/dapp-manager#97]
## Why?

We're using dapp-manager as a critical component of Golem Portal. We did already 2 load tests of our solution and we found out a performance decrease in the newest one. Digging deeper led to the discovery of worsened performance of the dapp-manager, which in turn reduces the number of concurrently served users.

## What was discovered?

We compared the dapp-manager versions 1a50eb355e685673f89596a3f523c5b819919822 and 0.1.0, which were used respectively in the first and second load test. We frequently "read state" of the dapp, so this command has been used as a benchmark.

### version 0.1.0

```
################################################
version 0.1.0
################################################

$ time dapp-manager read ed22767138684b1f89a5e006badfd1dd state --no-ensure-alive

{"nodes": {}, "app": "pending", "timestamp": "2023-03-09T12:23:41.054641+00:00"}
{"nodes": {"db": {"0": "pending"}}, "app": "starting", "timestamp": "2023-03-09T12:23:41.544249+00:00"}
{"nodes": {"db": {"0": "starting"}}, "app": "starting", "timestamp": "2023-03-09T12:23:54.604051+00:00"}
{"nodes": {"db": {"0": "running"}}, "app": "starting", "timestamp": "2023-03-09T12:23:58.387921+00:00"}
{"nodes": {"db": {"0": "running"}, "http": {"0": "pending"}}, "app": "starting", "timestamp": "2023-03-09T12:23:58.559123+00:00"}
{"nodes": {"db": {"0": "running"}, "http": {"0": "starting"}}, "app": "starting", "timestamp": "2023-03-09T12:24:13.022901+00:00"}
{"nodes": {"db": {"0": "running"}, "http": {"0": "running"}}, "app": "running", "timestamp": "2023-03-09T12:24:16.622155+00:00"}
{"nodes": {"db": {"0": "stopping"}, "http": {"0": "stopping"}}, "app": "stopping", "timestamp": "2023-03-09T13:24:17.610251+00:00"}
{"nodes": {"db": {"0": "stopping"}, "http": {"0": "stopping"}}, "app": "stopping", "timestamp": "2023-03-09T13:24:17.610281+00:00"}
{"nodes": {"db": {"0": "terminated"}, "http": {"0": "terminated"}}, "app": "terminated", "timestamp": "2023-03-09T13:24:18.609250+00:00"}
{"nodes": {"db": {"0": "terminated"}, "http": {"0": "terminated"}}, "app": "terminated", "timestamp": "2023-03-09T13:24:18.779278+00:00"}


________________________________________________________
Executed in  330.06 millis    fish           external
   usr time  321.36 millis    1.79 millis  319.56 millis
   sys time   14.62 millis    4.63 millis    9.99 millis
```

### version 1a50eb355e685673f89596a3f523c5b819919822

```
################################################
version 1a50eb355e685673f89596a3f523c5b819919822
################################################

$ time dapp-manager read ed22767138684b1f89a5e006badfd1dd state --no-ensure-alive

{"nodes": {}, "app": "pending", "timestamp": "2023-03-09T12:23:41.054641+00:00"}
{"nodes": {"db": {"0": "pending"}}, "app": "starting", "timestamp": "2023-03-09T12:23:41.544249+00:00"}
{"nodes": {"db": {"0": "starting"}}, "app": "starting", "timestamp": "2023-03-09T12:23:54.604051+00:00"}
{"nodes": {"db": {"0": "running"}}, "app": "starting", "timestamp": "2023-03-09T12:23:58.387921+00:00"}
{"nodes": {"db": {"0": "running"}, "http": {"0": "pending"}}, "app": "starting", "timestamp": "2023-03-09T12:23:58.559123+00:00"}
{"nodes": {"db": {"0": "running"}, "http": {"0": "starting"}}, "app": "starting", "timestamp": "2023-03-09T12:24:13.022901+00:00"}
{"nodes": {"db": {"0": "running"}, "http": {"0": "running"}}, "app": "running", "timestamp": "2023-03-09T12:24:16.622155+00:00"}
{"nodes": {"db": {"0": "stopping"}, "http": {"0": "stopping"}}, "app": "stopping", "timestamp": "2023-03-09T13:24:17.610251+00:00"}
{"nodes": {"db": {"0": "stopping"}, "http": {"0": "stopping"}}, "app": "stopping", "timestamp": "2023-03-09T13:24:17.610281+00:00"}
{"nodes": {"db": {"0": "terminated"}, "http": {"0": "terminated"}}, "app": "terminated", "timestamp": "2023-03-09T13:24:18.609250+00:00"}
{"nodes": {"db": {"0": "terminated"}, "http": {"0": "terminated"}}, "app": "terminated", "timestamp": "2023-03-09T13:24:18.779278+00:00"}


________________________________________________________
Executed in   70.26 millis    fish           external
   usr time   69.89 millis  210.00 micros   69.68 millis
   sys time    0.46 millis  458.00 micros    0.00 millis
```

## What would we expect?

Decrease of the execution time, preferably to below 100ms so that we can handle 3x more concurrent users on the same infrastructure.

**Comment by lucekdudek** [comment]
Update: I found the reason

https://github.com/golemfactory/dapp-manager/blob/main/dapp_manager/cli.py#L7

```python
from dapp_runner.log import LOG_CHOICES
```

takes 300 ms

I will look into it possible fixes

**Comment by lucekdudek** [comment]
Fixed in 0.1.1 release
